### PR TITLE
[#IOPID-257] Handle new login type on `acs` callback

### PR DESCRIPTION
**Pull request by gquadrati**
This PR depends on https://github.com/pagopa/io-spid-commons/pull/148

#### List of Changes

- Update io-spid-commons
- Enable new io-spid-commons functionality for handling additional data between login and acs steps.
- Add new FF management for LV functionality
- Create user's token based on LV FF

#### Motivation and Context

Enable new Fast Login management (based on FF).

#### How Has This Been Tested?

Unit test, io-mock

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -153,6 +153,7 @@ Those are all Environment variables needed by the application:
 | PORT                                   | The HTTP port the Express server is listening to                                  | int    |
 | REDIS_URL                              | The URL of a Redis instance                                                       | string |
 | TOKEN_DURATION_IN_SECONDS              | The number of seconds a session token is considered valid                         | int    |
+| LV_TOKEN_DURATION_IN_SECONDS           | The number of seconds a session token is considered valid if LV is enabled        | int    |
```
[the remark this answers is not in the document]
Comment (gquadrati) [reply README.md]: Thanks, done in [077a783](https://github.com/pagopa/io-backend/pull/1023/commits/077a7837dc34d47ea67530a6805c2be4edf54aa0)

File: src/utils/fastLogin.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export const LoginType = enumType<LoginTypeEnum>(LoginTypeEnum, "LoginType");
+
+export type AdditionalLoginPropsT = t.TypeOf<typeof AdditionalLoginProps>;
+export const AdditionalLoginProps = t.partial({ loginType: LoginType });
+
+export const acsRequestMapper = (
+  _req: express.Request
+): t.Validation<AdditionalLoginPropsT> =>
+  AdditionalLoginProps.decode({
+    loginType: _req.header("x-pagopa-login-type"),
```
[the remark this answers is not in the document]
Comment (gquadrati) [reply src/utils/fastLogin.ts]: Beccata 😄  Done in [077a783](https://github.com/pagopa/io-backend/pull/1023/commits/077a7837dc34d47ea67530a6805c2be4edf54aa0)

File: src/controllers/__tests__/authenticationController.test.ts [hunk 3]
```diff
@@ -925,6 +938,155 @@ describe("AuthenticationController#acs", () => {
...
+  );
+
+  it.each`
+    loginType                 | isUserEligible | expectedTtlDuration
+    ${LoginTypeEnum.LV}       | ${true}        | ${tokenDurationSecs}
+    ${LoginTypeEnum.LV}       | ${false}       | ${tokenDurationSecs}
+    ${LoginTypeEnum.STANDARD} | ${true}        | ${tokenDurationSecs}
+    ${LoginTypeEnum.STANDARD} | ${false}       | ${tokenDurationSecs}
+    ${undefined}              | ${true}        | ${tokenDurationSecs}
+    ${undefined}              | ${false}       | ${tokenDurationSecs}
```
[the remark this answers is not in the document]
Comment (gquadrati) [reply src/controllers/__tests__/authenticationController.test.ts]: Right, done in [077a783](https://github.com/pagopa/io-backend/pull/1023/commits/077a7837dc34d47ea67530a6805c2be4edf54aa0)

File: src/config.ts [hunk 4]
```diff
@@ -591,6 +591,14 @@ export const tokenDurationSecs: number = process.env.TOKEN_DURATION_IN_SECONDS
   : DEFAULT_TOKEN_DURATION_IN_SECONDS;
 log.info("Session token duration set to %s seconds", tokenDurationSecs);
 
+// Set default LV session duration
+const DEFAULT_LV_TOKEN_DURATION_IN_SECONDS = 60 * 15;
+export const lvTokenDurationSecs: number = process.env
+  .LV_TOKEN_DURATION_IN_SECONDS
+  ? parseInt(process.env.LV_TOKEN_DURATION_IN_SECONDS, 10)
+  : DEFAULT_LV_TOKEN_DURATION_IN_SECONDS;
```
[the remark this answers is not in the document]
Comment (gquadrati) [reply src/config.ts]: Thanks, done in [077a783](https://github.com/pagopa/io-backend/pull/1023/commits/077a7837dc34d47ea67530a6805c2be4edf54aa0)

File: src/controllers/authenticationController.ts [hunk 5]
```diff
@@ -175,6 +190,17 @@ export default class AuthenticationController {
       );
     }
 
+    // LV functionality is enable only if Lollipop is enabled.
+    // With FF set to BETA or CANARY, only whitelisted CF can use the LV functionality (the token TTL is reduced if login type is `LV`).
```
Comment (BurnedMarshal) [inline src/controllers/authenticationController.ts:194]: This seems to be wrong. If the FF is set to `CANARY` the LV is disabled for all the users on the current implementation:
```typescrypt
export const getIsUserElegibleForfastLogin = (
  betaTesters: ReadonlyArray<FiscalCode>,
  FF_FastLogin: FeatureFlag
) =>
  getIsUserEligibleForNewFeature<FiscalCode>(
    (fiscalCode) => betaTesters.includes(fiscalCode),
    (_fiscalCode) => false, // <----- CANARY
    FF_FastLogin
  );
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: src/controllers/authenticationController.ts (hunk 5 above)
Comment (BurnedMarshal) [reply src/controllers/authenticationController.ts:194]: Nope my bad, in the inner implementation canary includes betas user, sorry. But is not very clear is an hidden behaviour
```
case "CANARY":
        return isUserCanary(i) || isUserBeta(i);
```

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Pull request merged**